modules/host/contractmanager/storagefolder.go

```diff
@@ -315,13 +315,8 @@ func (cm *ContractManager) threadedFolderRecheck() {
 		// to the contract manager.
 		cm.wal.mu.Lock()
 		cm.sectorMu.Lock()
-		sfs := make([]*storageFolder, 0, len(cm.storageFolders))
-		for _, sf := range cm.storageFolders {
-			sfs = append(sfs, sf)
-		}
-		cm.sectorMu.Unlock()
 
-		for _, sf := range sfs {
+		for _, sf := range cm.storageFolders {
```

Comment:
Did this cause issues as well? The reason I copied the folders first was to avoid opening a file while holding `sectorMu` every second we trigger the recheck.

But I guess if we only do this once per storage folder and the remaining times we just check the atomic and skip opening the file it's fine.

Reply:
It wasn't causing problems, but it seemed like the safer route after removing `sectorMu` from `loadSectorLocations`. I'll put the copy back and also wrap `loadSectorLocations`.

Reply:
Do the changes look okay with what you were going for?